### ISSUE TITLE
Ursula submitted commitToNextPeriod() transaction with less than 50k gas limit causing it to fail

### ISSUE THREAD

**Issue by svenski123** [nucypher/nucypher#2406]
Ursula mainnet worker running v4.1.0 submitted a commitToNextPeriod() transaction that failed due to an Out of Gas condition.

The gas limit on the failing transaction was 49,691 which seems unusually low.

Seventeen minutes later the worker submitted a new commitToNextPeriod() transaction that succeeded - that one had a gas limit of 280,248 of which 169,534 was actually used.

Unclear why the first transaction had such a low gas limit, perhaps there is a transient issue with gas estimation.

Failed transaction: https://etherscan.io/tx/0x7f418a89fb200e1dafa9c15b4e8d3265fd25ccfd1595130660075497e3752728
Successful transaction: https://etherscan.io/tx/0xb040b25d00d72b2325b367f43f5e0e887e0a517df1a4779096a642c0ae17ab34

Console log output in Ursula was not terribly enlightening:
```
Broadcasting COMMITTONEXTPERIOD Transaction (0.001017942110745368 ETH @ 20.485442248 gwei)...
Broadcasting COMMITTONEXTPERIOD Transaction (0.00574356834034656 ETH @ 20.49459172 gwei)...
```

**Comment by svenski123** [comment]
This happened again, Ursula submitted this transaction with a too small gas limit that ran out of gas:
https://etherscan.io/tx/0x54205252edb40d6e6317cf8add55d51835538ee49a9f8770bb6b858db870f47e

It then submitted a new transaction with a normal gas limit and a higher gas price which succeeded:
https://etherscan.io/tx/0xa25091851a8e202ed4194bce8899926896645b86548878dadefe939cbceb8ed0

**Comment by cygnusv** [comment]
This issue should be solved by #2395. The problem is that the logic to check for what's the current period should happen before the logic to track pending transactions

**Comment by cygnusv** [comment]
Rescoped #2395 to do something else. This problem is addressed now by
